### ISSUE TITLE
Return value is not referentially stable

### ISSUE THREAD

**Issue by merisbahti** [mweststrate/use-st8#13]
The return value here:
https://github.com/mweststrate/use-st8/blob/743172aff4a97eb63e311a9985484948236db373/src/index.ts#L17

Is not referentially stable.

Should it be wrapped in a memo perhaps?

**Comment by mweststrate** [comment]
yeah looks like it! not memo but useState or a ref, otherwise it is still
not stable :)

**Comment by mweststrate** [comment]
Not guaranteed I mean

**Comment by merisbahti** [comment]
So how would the ref solution work to make it stable? I'm curious.

Would we put the states current value in the ref, then in the function return the refs current value?

Wonder if that's CM compatible!

**Comment by merisbahti** [comment]
I just realised that if this value was referentially stable, it would cause bugs.

imagine the simple scenario where we have a `Parent`, that initializes state `const st8 = useSt8(5)`, and a child which takes the `st8` as a prop.

If the st8 value is updated (via parent or child) and the child is wrapped in `React.memo` - the child would opt out from rendering, because its `st8` prop is still the same reference.

Closing this as that would break this lib :)
